sub-packages/bionemo-esm2/src/bionemo/esm2/testing/eval.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import gc
+from pathlib import Path
+
+import torch
+from megatron.core.transformer.module import Float16Module
+
+from bionemo.core.utils.dtypes import PrecisionTypes, get_autocast_dtype
+from bionemo.esm2.data import tokenizer
+from bionemo.esm2.model.model import ESM2Config
+from bionemo.testing import megatron_parallel_state_utils
```

Comment:
I will have to move `megatron_parallel_state_utils` from `bionemo.testing` for this to work.
CC: @jstjohn @pstjohn

Reply:
Yeah I had this same issue and almost put a thread together for it. The other option (which honestly might be better) is just to move the context manager outside of `eval_esm2`) and just ensure that `eval_esm2` is called inside one. Thats what I do with `assert_model_equivalence`

Reply:
I.e., I think we can call into `bionemo.testing` in the example notebooks without tach yelling at us, and it makes the creation of the megatron context more explicit

Reply:
I figured we might run into cases where we iterate on a dataloader, so I separated the setup/teardown to avoid configuring the model every time we call forward. 

Now I can also add a method to check the input to eval and tokenize if they are sequences. But I don't think that would look clean

Reply:
yeah, I moved the context manager outside and into the notebooks, no complain from tach!